homeassistant/components/egps/switch.py

```diff
@@ -0,0 +1,85 @@
+"""Switch implementation for EGPS Platform."""
+
+from typing import Any
+
+from pyegps import __version__ as PYEGPS_VERSION
+from pyegps.exceptions import EgpsException
+from pyegps.powerstrip import PowerStrip
+
+from homeassistant.components.switch import SwitchDeviceClass, SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add EGPS sockets for passed config_entry in HA."""
+    powerstrip: PowerStrip = hass.data[DOMAIN][config_entry.entry_id]
+
+    async_add_entities(
+        [
+            EGPowerStripSocket(powerstrip, socket)
+            for socket in range(powerstrip.numberOfSockets)
+        ],
+        update_before_add=True,
+    )
+
+
+class EGPowerStripSocket(SwitchEntity):
+    """Represents a socket of an Energenie-Socket-Strip."""
+
+    _attr_device_class = SwitchDeviceClass.OUTLET
+    _attr_should_poll = True
+    _attr_has_entity_name = True
+    _attr_translation_key = "socket"
+
+    def __init__(self, dev: PowerStrip, socket: int) -> None:
+        """Initiate a new socket."""
+        self._dev = dev
+        self._socket = socket
+        self._state = STATE_OFF
+        self._attr_translation_placeholders = {"socket_id": str(socket)}
+
+        self._attr_unique_id = f"{dev.device_id}_{socket}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, dev.device_id)},
+            name=dev.name,
+            manufacturer=dev.manufacturer,
+            model=dev.name,
+            sw_version=PYEGPS_VERSION,
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return True if socket is on, else False."""
+        return self._state == STATE_ON
```

Comment:
Instead you can store the bool in `_attr_is_on`, would make it a bit cleaner

Reply:
Instead of assigning STATE ON and STATE OFF to _state based on the device status, just assign the device status to _attr_is_on

Reply:
that's what I did, but the tests won't pass anymore. In the tests all states are read as 'off' now,
what do I miss?
```python
def update(self) -> None:
        """Read the current state from the device."""
        try:
            self._attr_is_on = self._dev.get_status(self._socket) == STATE_ON
        except EgpsException as err:
            raise HomeAssistantError(f"Couldn't access USB device: {err}") from err
```

Reply:
Did you remove the `is_on` property?

Reply:
yes

Reply:
Also, doesn't that device function return a Boolean? Looking at the current code you never compare with STATE_ON, so that sounds invalid

Reply:
ohhh yes you are right, thanks a lot! sorry ;)